### ISSUE TITLE
Error setting up physicell studio: error message  'QSpinBox' object has no attribute 'textChanged'

### ISSUE THREAD

**Issue by sessler** [MathCancer/PhysiCell#312]
I was trying to set up physicell + phisicell stugio following the tutorial youtube video Windows PhysiCell ecosystem install guide July 2023. When setting up the studio 
(base) C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell>python C:\Users\sessl\Downloads\get_studio.py
v2.40.0 - bug fixes and improvements to GUI and/or code base
['v2.40.0', '-', 'bug', 'fixes', 'and', 'improvements', 'to', 'GUI', 'and/or', 'code', 'base']
vnum= v2.40.0
remote_url= https://github.com/PhysiCell-Tools/PhysiCell-Studio/archive/refs/tags/v2.40.0.zip
downloaded version  v2.40.0  to  Studio.zip
unzip-ing  Studio.zip
renamed 'PhysiCell-Studio-2.40.0' to 'studio'
 and tried to run it I got error message
(base) C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell>python studio/bin/studio.py
args= Namespace(bare=False, three=False, tensor=False, rules=False, skip_validate=False, nanohub=False, config=None, exec=None, biwt_flag=False)
unknown= []
PhysiCellXMLCreator(): self.nanohub_flag=  False
  platform.system().lower()=windows, platform.machine()=AMD64
Could not parse application stylesheet
self.current_dir =  C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell
studio: (default) self.config_file = C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\config\PhysiCell_settings.xml
studio: (default) self.tree = <xml.etree.ElementTree.ElementTree object at 0x0000024DF0849E80>
studio: (default) self.xml_root = <Element 'PhysiCell_settings' at 0x0000024DF087A590>
Traceback (most recent call last):
  File "C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio\bin\studio.py", line 1607, in <module>
    main()
  File "C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio\bin\studio.py", line 1555, in main
    ex = PhysiCellXMLCreator(config_file, studio_flag, skip_validate_flag, rules_flag, model3D_flag, tensor_flag, exec_file, nanohub_flag, is_movable_flag, pytest_flag, biwt_flag
  File "C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio\bin\studio.py", line 215, in __init__
    self.config_tab = Config(self)
  File "C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio\bin\config_tab.py", line 206, in __init__
    self.day_spin_box.textChanged.connect(self.day_value_changed_str)
AttributeError: 'QSpinBox' object has no attribute 'textChanged'

Thank you for your help
Judit Sessler

**Comment by rheiland** [comment]
Sorry for this and thanks for reporting the issue. It seemed to run OK on my Windows 11 using:
```
PS C:\Users\randy\PhysiCell-Studio-2.40.0\PhysiCell-Studio-2.40.0> python
Python 3.10.10 | packaged by Anaconda, Inc. | (main, Mar 21 2023, 18:39:17) [MSC v.1916 64 bit (AMD64)] on win32

>>> from PyQt5.QtCore import QT_VERSION_STR, PYQT_VERSION_STR
>>> print("Qt: v", QT_VERSION_STR, "\tPyQt: v", PYQT_VERSION_STR)
Qt: v 5.15.2    PyQt: v 5.15.7
```
but maybe it's somewhat irrelevant. A quick search seems to indicate that the QSpinBox's textChanged method may be being replaced with valueChanged? If you wouldn't mind, could you download the attached file and copy it into your studio's \bin folder, renaming it `config_tab.py` then re-run as you did before to see if things improve?
[config_tab.txt](https://github.com/user-attachments/files/17412789/config_tab.txt)

**Comment by sessler** [comment]
Thanks for the reply, but now the error message changed to:

(base)
C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio>python
bin/studio.py
args= Namespace(bare=False, three=False, tensor=False, rules=False,
skip_validate=False, nanohub=False, config=None, exec=None, biwt_flag=False)
unknown= []
PhysiCellXMLCreator(): self.nanohub_flag=  False
  platform.system().lower()=windows, platform.machine()=AMD64
Could not parse application stylesheet
self.current_dir =
 C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio
studio: (default) self.config_file =
C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio\config\PhysiCell_settings.xml
studio: (default) self.tree = <xml.etree.ElementTree.ElementTree object at
0x000002724DB89370>
studio: (default) self.xml_root = <Element 'PhysiCell_settings' at
0x000002724DB8AD60>
Traceback (most recent call last):
  File
"C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio\bin\studio.py",
line 1607, in <module>
    main()
  File
"C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio\bin\studio.py",
line 1555, in main
    ex = PhysiCellXMLCreator(config_file, studio_flag, skip_validate_flag,
rules_flag, model3D_flag, tensor_flag, exec_file, nanohub_flag,
is_movable_flag, pytest_flag, biwt_flag
  File
"C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio\bin\studio.py",
line 215, in __init__
    self.config_tab = Config(self)
  File
"C:\Users\sessl\OneDrive\Documents\PhysiCell_V.1.14.0\PhysiCell\studio\bin\config_tab.py",
line 330, in __init__
    self.random_seed_gp.idToggled.connect(self.random_seed_gp_cb)
AttributeError: 'QButtonGroup' object has no attribute 'idToggled'

Randy Heiland ***@***.***> ezt írta (időpont: 2024. okt. 17.,
Cs, 15:01):

**Comment by sessler** [comment]
[image: image.png]
I have just realised, PyQt5 is missing. How can I remedy this?



Randy Heiland ***@***.***> ezt írta (időpont: 2024. okt. 17.,
Cs, 17:12):

**Comment by rheiland** [comment]
Strange. If you meant to share a screen image of your Python version, it didn't show. If it's easier, please email info to heiland@iu.edu 

You might try to run the command `pip3 install PyQt5` from your terminal. If it complains that `pip3` is not found, try `pip`.

**Comment by rheiland** [comment]
But if PyQt5 was really missing, I think you'd get a very different error message. So I'm confused.

**Comment by sessler** [comment]
I tried the install, it worked, though I got some error messages:
Installing collected packages: PyQt5-Qt5, PyQt5-sip, PyQt5
ERROR: pip's dependency resolver does not currently take into account all
the packages that are installed. This behaviour is the source of the
following dependency conflicts.
spyder 5.2.2 requires pyqtwebengine<5.13, which is not installed.
spyder 5.2.2 requires pyqt5<5.13, but you have pyqt5 5.15.11 which is
incompatible.
Successfully installed PyQt5-5.15.11 PyQt5-Qt5-5.15.2 PyQt5-sip-12.15.0

but now studio started up fine THANK YOU very much for helping me out,and
so fast.


Randy Heiland ***@***.***> ezt írta (időpont: 2024. okt. 17.,
Cs, 17:40):
